### PR TITLE
Refactor metrics

### PR DESCRIPTION
**Pull request by SergeyTsimfer**
This PR proposes to:

- Completely refactor functions of metric evaluation, as well as methods of cutting cube amplitudes

- Move functions from `utils.py` to separate files: `horizon.py`, `labels_utils.py`. Also `_const.py` is created

- `create_mask` action is now capable of loading the exact chosen horizons

- better plot functions

- add some of the missing docstrings

### CODE REVIEW

**Comment by SergeyTsimfer** [comment]
[refactor metrics](https://trello.com/c/CSJbwoag/1-refactor-metrics)

File: seismiqb/src/labels_utils.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        df.sort_values(['iline', 'xline'], inplace=True)
+
+    data = df.loc[:, order]
+    if transform:
+        data = data.apply(transform)
+    data.to_csv(path_save, sep=' ', index=False, header=False)
+
+
+def read_point_cloud(paths, names=None, order=None, **kwargs):
+    """ Read point cloud of labels from files using pandas.
```
Comment (dmylzenova) [inline seismiqb/src/labels_utils.py]: I think a few words on the expected format of the pc file (at least a column sep)  would be appreciated. It is not clear from the code why exactly do you need convert_point_cloud function.ax

File: seismiqb/src/labels_utils.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    with open(path) as file:
+        line = file.readline().split(' ')
+    if len(line) == 3:
+        return ['iline', 'xline', 'height']
+    return ['iline', 'xline', 's_point', 'e_point']
+
+
+@njit
+def filter_point_cloud(point_cloud, filtering_matrix, ilines_offset, xlines_offset):
+    """ Remove entries corresponding to zero traces.
```
Comment (dmylzenova) [inline seismiqb/src/labels_utils.py]: not necessarily zero traces

File: seismiqb/src/plot_utils.py [hunk 3]
```diff
@@ -164,9 +177,24 @@ def _to_img(data, order_axes=None, convert=False, normalize=False):
 
 
 def plot_slide(dataset, *components, idx=0, n_line=0, plot_mode='overlap', mode='iline', **kwargs):
-    """ Plot full slide of the given cube on the given iline. """
+    """ Plot full slide of the given cube on the given n_line.
```
Comment (dmylzenova) [inline seismiqb/src/plot_utils.py]: can we add point and crop_shape to this function?

File: seismiqb/src/utils.py [hunk 4]
```diff
@@ -626,171 +242,30 @@ def update_minmax(array, val_min, val_max, matrix, il, xl, ilines_offset, xlines
...
-            slide -= geom.value_min
-            slide *= (1 / scale_val)
-        # here we can insert even more transforms!
-        background, depth_map = _update(background, depth_map, slide, labels,
-                                        x_len, il, i_offset, x_offset,
-                                        low, high, horizon_idx, offset)
-    background = np.squeeze(background)
-    return background, depth_map
-
+def compute_running_mean(x, kernel_size):
```
Comment (dmylzenova) [inline seismiqb/src/utils.py]: this function was used in the old metrics computation and is not used anywhere now.

File: seismiqb/src/utils.py [hunk 5]
```diff
@@ -626,171 +242,30 @@ def update_minmax(array, val_min, val_max, matrix, il, xl, ilines_offset, xlines
...
-            if x < x_range-1:
-                trace_4 = data[i, x+1, :]
-                if not np.max(trace_4) == np.min(trace_4):
-                    s += np.corrcoef(trace, trace_4)[0, 1]
-                    c += 1
-
-            if c != 0:
-                corrs[i, x] = s / c
-    return corrs
+def _compute_running_mean_jit(x, kernel_size, cumsum):
```
Comment (dmylzenova) [inline seismiqb/src/utils.py]: but it looks better now)

File: seismiqb/src/utils.py [hunk 6]
```diff
@@ -626,171 +243,30 @@ def update_minmax(array, val_min, val_max, matrix, il, xl, ilines_offset, xlines
...
+    """ Jit accelerated running mean. """
+    #pylint: disable=invalid-name
```
Comment (roman-kh) [inline seismiqb/src/utils.py]: Add variable name regex to pylintrc

**Pull request merged**